core/src/test/java/com/sequenceiq/cloudbreak/service/stack/ServiceStatusCheckerLogLocationDecoratorTest.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.cluster.status.ExtendedHostStatuses;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.cloudbreak.domain.stack.Component;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.service.ComponentConfigProviderService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+import com.sequenceiq.common.api.telemetry.model.Logging;
+import com.sequenceiq.common.api.telemetry.model.Telemetry;
+
+@ExtendWith(MockitoExtension.class)
+class ServiceStatusCheckerLogLocationDecoratorTest {
+
+    @Mock
+    private ComponentConfigProviderService componentConfigProviderService;
+
+    @Mock
+    private Component component;
+
+    @Mock
+    private Json attributes;
+
+    @Mock
+    private Telemetry telemetry;
+
+    @Mock
+    private Logging logging;
+
+    @Mock
+    private InstanceMetadataView instanceMetadataView;
+
+    @Mock
+    private ExtendedHostStatuses extendedHostStatuses;
+
+    @Mock
+    private StackDto stackDto;
+
+    @InjectMocks
+    private ServiceStatusCheckerLogLocationDecorator underTest;
+
+    @BeforeEach
+    void setup() throws IOException {
+        lenient().when(componentConfigProviderService.getComponent(anyLong(), any(), any())).thenReturn(component);
+        lenient().when(component.getAttributes()).thenReturn(attributes);
+        lenient().when(attributes.get(Telemetry.class)).thenReturn(telemetry);
+        lenient().when(telemetry.getLogging()).thenReturn(logging);
+        lenient().when(logging.getStorageLocation()).thenReturn("storageLocation");
+        lenient().when(stackDto.getId()).thenReturn(1L);
+    }
+
+    @Test
+    void testDecorateWithServiceFailure() {
+        when(extendedHostStatuses.hasHostUnhealthyServices(any())).thenReturn(true);
+        Map<InstanceMetadataView, Optional<String>> result = underTest.decorate(getInstanceMetadataWithReason(), extendedHostStatuses, stackDto);
+        assertEquals("DefaultMessage. Please check the logs at the following location: storageLocation",
+                result.entrySet().stream().findFirst().flatMap(Map.Entry::getValue).orElse(""));
+        verify(componentConfigProviderService, times(1)).getComponent(anyLong(), any(), any());
+    }
+
+    @Test
+    void testDecorateWithNoServiceFailure() {
+        when(extendedHostStatuses.hasHostUnhealthyServices(any())).thenReturn(false);
+        Map<InstanceMetadataView, Optional<String>> result = underTest.decorate(getInstanceMetadataWithReason(), extendedHostStatuses, stackDto);
+        assertEquals("DefaultMessage.",
+                result.entrySet().stream().findFirst().flatMap(Map.Entry::getValue).orElse(""));
+        verify(componentConfigProviderService, times(0)).getComponent(anyLong(), any(), any());
+    }
```

Comment:
There seems to be no test case for the `InstanceStatus.SERVICES_UNHEALTHY == instanceMetadataView.getInstanceStatus()` scenario.

Reply:
good catch!